features/FEATURE_BLE/targets/TARGET_CORDIO/driver/CordioHCIDriver.cpp

```diff
@@ -144,25 +144,17 @@ void CordioHCIDriver::handle_reset_sequence(uint8_t *pMsg)
                 /* parse and store event parameters */
                 BdaCpy(hciCoreCb.bdAddr, pMsg);
 
-                ble::address_t static_address;
-
-                if (get_random_static_address(static_address)) {
-                    // note: will send the HCI command to send the random address
-                    cordio::BLE::deviceInstance().getGap().setAddress(
-                        BLEProtocol::AddressType::RANDOM_STATIC,
-                        static_address.data()
-                    );
-                } else {
-                    /* send next command in sequence */
-                    HciLeReadBufSizeCmd();
-                }
-                break;
-            }
+                ble::address_t dummy;
+                /* public address cannot be set so this does not create an HCI command */
+                cordio::BLE::deviceInstance().getGap().setAddress(
+                    BLEProtocol::AddressType::PUBLIC,
+                    dummy
+                );
 
-            case HCI_OPCODE_LE_SET_RAND_ADDR:
```

Comment:
That is still needed, a call to setAddress will send an HCI command to set the address; I know this is ugly.

Reply:
set address doesn't create a HCI event for public, also turns out we don't need to call that because it's already public at init